### PR TITLE
fixed issues with sorting and axes

### PR DESCRIPTION
**Pull request by helen508light**

### CODE REVIEW

**Comment by ignatvilesov** [comment]
Merge conflicts. Please resolve all of them

File: package.json [hunk 1]
```diff
@@ -49,7 +49,13 @@
...
     "karma-sourcemap-loader": "0.3.7",
     "karma-typescript-preprocessor": "0.3.1",
-    "powerbi-visuals-tools": "1.13.1",
+    "lodash": "4.16.2",
+    "powerbi-models": "^1.0.2",
+    "powerbi-visuals-tools": "1.13.0",
+    "powerbi-visuals-utils-chartutils": "1.5.1",
+    "powerbi-visuals-utils-dataviewutils": "1.2.0",
+    "powerbi-visuals-utils-formattingutils": "3.0.1",
+    "powerbi-visuals-utils-interactivityutils": "3.1.0",
```
Comment (ignatvilesov) [inline package.json]: These added dependencies are already in `dependencies`. Please remove useless from `devDependencies`

File: pbiviz.json [hunk 2]
```diff
@@ -21,6 +21,9 @@
     "node_modules/jquery/dist/jquery.min.js",
     "node_modules/lodash/lodash.min.js",
     "node_modules/d3/d3.js",
+    "node_modules/powerbi-visuals-utils-typeutils/lib/index.js",
+    "node_modules/powerbi-visuals-utils-dataviewutils/lib/index.js",
+    "node_modules/powerbi-visuals-utils-formattingutils/lib/index.js",
```
Comment (ignatvilesov) [inline pbiviz.json]: These three js files are already included.

File: src/visual.ts [hunk 3]
```diff
@@ -412,8 +413,30 @@ module powerbi.extensibility.visual {
                 this.data.dateValues,
                 (dateValue: DateValue) => dateValue.value);
 
-            const minDate: number = extentDate[0];
-            const maxDate: number = extentDate[1] + (extentDate[1] - extentDate[0]) * LineDotChart.dateMaxCutter;
+            const columnFormattingFn = (index: number, dataType: valueType): any => {
```
Comment (ignatvilesov) [inline src/visual.ts]: Let's not to create anonymous function. It'd be better to convert it to a method and use bind in constructor like we do in React

File: src/visual.ts [hunk 4]
```diff
@@ -412,8 +413,30 @@ module powerbi.extensibility.visual {
...
+                if (dataType.dateTime) {
+                    return this.data.dateColumnFormatter.format(new Date(index));
+                }
+                else if (dataType.text) {
+                    return this.data.dateValues[index].label;
+                }
+                return this.data.dateColumnFormatter.format(index);
+            };
+
+            const valueFormattingFn = (index: number, dataType: valueType): any => {
```
Comment (ignatvilesov) [inline src/visual.ts]: Let's not to create anonymous function. It'd be better to convert it to a method and use bind in constructor like we do in React

File: src/visual.ts [hunk 5]
```diff
@@ -447,29 +462,44 @@ module powerbi.extensibility.visual {
             );
 
             this.xAxisProperties.formatter = this.data.dateColumnFormatter;
+            let yMin = this.data.yMinValue,
+                yMax = this.data.yMaxValue;
```
Comment (ignatvilesov) [inline src/visual.ts]: Let use an own `let` for each declaration:

```typescript
let yMin = this.data.yMinValue;
let yMax = this.data.yMaxValue;
```

File: src/visual.ts [hunk 6]
```diff
@@ -447,29 +462,44 @@ module powerbi.extensibility.visual {
...
+            if (yMax === yMin) {
```
Comment (ignatvilesov) [inline src/visual.ts]: Can you add a coment to make it clear what is going on here

File: src/visual.ts [hunk 7]
```diff
@@ -1079,9 +1106,8 @@ module powerbi.extensibility.visual {
 
             const unformattedDate: string | number = dataPoint.dateValue.label || dataPoint.dateValue.value;
 
-            const valueFormatterLocalized = valueFormatter.create({ cultureSelector: this.hostService.locale });
-            const formattedDate: string = this.data.dateColumnFormatter.format(unformattedDate);
-            const formattedValue: string = this.data.dataValueFormatter.format(valueFormatterLocalized.format(dataPoint.value));
+            const formattedDate: string = this.data.dateColumnFormatter.format(unformattedDate),
+                formattedValue: string = this.data.dataValueFormatter.format(dataPoint.value);
```
Comment (ignatvilesov) [inline src/visual.ts]:
```typescript
const formattedDate: string = this.data.dateColumnFormatter.format(unformattedDate);
const formattedValue: string = this.data.dataValueFormatter.format(dataPoint.value);
```

File: src/visual.ts [hunk 8]
```diff
@@ -375,7 +375,8 @@ module powerbi.extensibility.visual {
             };
 
             const dataValueFormatter: IValueFormatter = valueFormatter.create({
-                format: valueFormatter.getFormatStringByColumn(valuesColumn.source)
+                format: valueFormatter.getFormatStringByColumn(valuesColumn.source, true) || "#",
```
Comment (ignatvilesov) [inline src/visual.ts]: Are these changes covered by UTs?

File: src/visual.ts [hunk 9]
```diff
@@ -447,29 +462,44 @@ module powerbi.extensibility.visual {
...
                 isVertical: true,
-                useTickIntervalForDisplayUnits: true
+                useTickIntervalForDisplayUnits: true,
+                getValueFn: valueFormattingFn
             });
 
             this.yAxis2Properties = AxisHelper.createAxis({
                 pixelSpan: effectiveHeight,
-                dataDomain: [this.data.yMinValue, this.data.sumOfValues],
+                dataDomain: [yMin, yMax],
```
Comment (ignatvilesov) [inline src/visual.ts]: IF we are going to publish it we should increase versions and describe changes in CHANGELOG.md

File: src/visual.ts (hunk 8 above)
Comment (uve) [reply src/visual.ts]: please add unit-tests

**Pull request closed without merging**